src/System.Private.Xml/src/System/Xml/XmlDownloadManagerAsync.cs

```diff
@@ -34,22 +34,22 @@ internal partial class XmlDownloadManager
         private async Task<Stream> GetNonFileStreamAsync(Uri uri, ICredentials credentials, IWebProxy proxy,
             RequestCachePolicy cachePolicy)
         {
-            HttpClient client = new HttpClient();
-            HttpRequestMessage req = new HttpRequestMessage(HttpMethod.Get, uri);
-
-            lock (this)
+            WebRequest req = WebRequest.Create(uri);
+            if (credentials != null)
             {
-                if (_connections == null)
-                {
-                    _connections = new Hashtable();
-                }
+                req.Credentials = credentials;
+            }
+            if (proxy != null)
+            {
+                req.Proxy = proxy;
+            }
+            if (cachePolicy != null)
+            {
+                req.CachePolicy = cachePolicy;
             }
 
-            HttpResponseMessage resp = await client.SendAsync(req).ConfigureAwait(false);
-
-            Stream respStream = new MemoryStream();
-            await resp.Content.CopyToAsync(respStream);
-            return respStream;
+            WebResponse resp = await Task<WebResponse>.Factory.FromAsync(req.BeginGetResponse, req.EndGetResponse, null).ConfigureAwait(false);
+            return resp.GetResponseStream();
```

Comment:
The old version was copying into a memory stream and returning that memory stream.  Why was that being done?  Is that needed here?  If that is done, please also dispose of the WebResponse and the response Stream after you've copied it to the MemoryStream (such disposal should have been done in the old case as well).